### ISSUE TITLE
Q# language server does not work in VS Code when .NET SDK is unavailable

### ISSUE THREAD

**Issue by cgranade** [microsoft/qsharp-compiler#1235]
**Describe the bug**

When using the VS Code extension for Q#, language server functionality is not available unless the `dotnet` command is present. In particular, when working with Q# + Python workflows using the `qsharp` conda package, the .NET SDK is not typically installed.

**To Reproduce**

1. From a new Windows Sandbox instance, install VS Code.
2. Launch VS Code.
2. From the extensions activity, install the Quantum Development Kit extension (version v0.20.2110171573).
2. From the command palette, run "Q#: Create new project," and observe first false positive error message:

![image](https://user-images.githubusercontent.com/31516/140115824-e3eb07c6-b4e6-42a2-b7d7-2d3323295c0e.png)

5. Continue in the quick prompt by selecting "Standalone application," choosing a project, and then opening the project in a new window by selecting "Open new project..."

![image](https://user-images.githubusercontent.com/31516/140117550-22de3fcf-82d6-4e9a-b81d-6060a9af5b6b.png)

6. Open `Program.qs` in the new window, and observe two resulting errors:
![image](https://user-images.githubusercontent.com/31516/140118687-51f84813-49ae-402e-bcb8-8c9b2b6e5792.png)

7. Make edits in Q# source and note that only basic diagnostics are provided (e.g.: syntax errors), and that no hover information / signature help is provided.

**Expected behavior**

The self-contained Q# language server installed by the VS Code extension should run correctly when the `dotnet` command is missing.

**Screenshots**

See **§To Reproduce**, above.

**System information**

See **§To Reproduce**, above.

**Additional context**

This is filed as a bug and not a feature request, as the dependency appears to be inadvertent — the Q# language server used to work in self-contained mode without .NET, and the Q# + Python install docs (https://docs.microsoft.com/azure/quantum/install-python-qdk?tabs=tabid-conda#choose-your-ide) do not instruct users to install the .NET SDK unless using the `dotnet`+`pip` install method.

**Comment by swernli** [comment]
By way of additional info, here is the error that is logged to the Q# Language Extension's output window:
```
[Error - 12:12:08 PM] 
=============================
Microsoft.Quantum.QsCompiler.QsCompilerException: error while running queued task
FileNotFoundException: Could not load file or assembly 'Microsoft.Build, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
Stack trace: 
   at Microsoft.Quantum.QsLanguageServer.EditorState.QsProjectLoader(Uri projectFile, ProjectInformation& info)
   at Microsoft.Quantum.QsCompiler.CompilationBuilder.ProjectManager.<>c__DisplayClass12_0.<LoadProjectsAsync>b__1() in D:\a\1\s\submodules\qsharp-compiler\src\QsCompiler\CompilationManager\ProjectManager.cs:line 920
   at Microsoft.Quantum.QsCompiler.QsCompilerError.RaiseOnFailure(Action action, String header) in D:\a\1\s\submodules\qsharp-compiler\src\QsCompiler\DataStructures\ErrorHandling.fs:line 60

   at Microsoft.Quantum.QsCompiler.QsCompilerError.Raise(String message, Exception innerException) in D:\a\1\s\submodules\qsharp-compiler\src\QsCompiler\DataStructures\ErrorHandling.fs:line 29
   at Microsoft.Quantum.QsCompiler.QsCompilerError.RaiseOnFailure(Action action, String header) in D:\a\1\s\submodules\qsharp-compiler\src\QsCompiler\DataStructures\ErrorHandling.fs:line 63
   at Microsoft.Quantum.QsCompiler.CompilationBuilder.ProcessingQueue.<>c__DisplayClass4_0.<ProcessingTaskAsync>b__0() in D:\a\1\s\submodules\qsharp-compiler\src\QsCompiler\CompilationManager\ProcessingQueue.cs:line 42
=============================

[Info  - 12:12:09 PM] The file c:\Users\WDAGUtilityAccount\Desktop\qsharptest\Program.qs is not associated with a compilation unit. Only syntactic diagnostics are generated.
```

**Comment by swernli** [comment]
From what I can tell, this may never have worked all the way, and is a gap in our support for non-.NET scenarios. When searching for dependencies of the current project, the Language server expects to load and parse the csproj file and navigate the dependency tree. To do that, it must use the `Microsoft.Build` package which it finds via the `Microsoft.Build.Locator`, invoked here: https://github.com/microsoft/qsharp-compiler/blob/a4ef30735d856069f92ef33e00e92adcf433f49b/src/QsCompiler/LanguageServer/Program.cs#L120. If that line throws, it means the server was unable to locate any locally installed MSBuild, either from .NET Core SDK or Visual Studio, and won't be able to parse dependencies. The end result is that the extension operates in a significantly reduced capacity, providing only syntax coloring and basic feedback but no type checking, target capability checks, or any other features. In that sense, the full experience is indeed dependent on finding MSBuild. Along the way, the attempt to load the EditorState features results in an exception, which is the source of the "Q# Language Server has encountered an error..." message and the stack above. Some ideas:
- Add state to the extension that prevents it from trying to load projects when it hasn't located MSBuild.
This will at least prevent the error popup, and could provide the opportunity for a warning that mentions the reduced features. This is probably the right short term fix, but is not really sufficient for long term user experience.

- Distribute MSBuild binaries with the Language Server.
This is generally not recommended by the .NET team and would be a pain to manage, but would allow the Language Server to work in more environments.

- Have the VSCode extension download any needed MSBuild components.
I'm not sure this is even possible, but if the extension could manage this it might give a better end-user experience without forcing the Language Server itself to carry the full set of MSBuild binaries.

- Update docs and requirements to say either .NET SDK or Visual Studio Build Tools are required to get full editor access.
Not a great solution, especially given we want to support the condo/iqsharp route where a user should not have to install the .NET SDK, but may make sense in light of the switch to .NET 6 and the other breaking changes that entails (update to VS 2022, forced update of other extensions like OmniSharp, etc).

@cgranade, @ricardo-espinoza, @bettinaheim for FYI.
